### DISCUSSION TITLE
Changing the display of grammatical compound forms

### DISCUSSION THREAD

**Discussion by enellis** [birchill/10ten-ja-reader#1819] [Ideas]
Currently, many grammatical compound forms are displayed as they were one single form, e.g. "< polite past negative". To better reflect what is happening linguistically, I had the idea to change it to something like "< polite < negative < past".

Current state of [discussion](https://github.com/birchill/10ten-ja-reader/pull/1814#issuecomment-2161383642):
> Thinking about it, I’d like to change the "reason" field to an array to eliminate the need for compound forms like PolitePast, PolitePastNegative, etc. This would be an user-facing change because for example, 食べませんでした would be parsed as `< polite < past < negative` (or better `< polite < negative < past` as this reflects the grammatical structure more closely) instead of `< polite past negative`.
>
> &mdash; <cite>Me</cite>

> I think "polite past negative" might be preferable to most Japanese learners. Even if "< polite < negative < past" is more accurate linguistically, I think most learners learn ～ませんでした as a set and are used to thinking of it as such so I think "< polite < negative < past" might actually be less useful.
>
> &mdash; <cite>birtles</cite>

> Yeah, I can see that. Removing the compound forms was just a quick idea which isn't really necessary.
>
> &mdash; <cite>Me</cite>

> I'm not too worried which form we end up with. My initial reaction to "< polite < negative < past" was just that it looked complicated for something that is reasonably common but maybe that's more to do with the "<" delimiters. Maybe we should consider another way of combining the reasons?
>
> &mdash; <cite>birtles</cite>

> Maybe it would help already making them a little bit smaller, like "‹ polite ‹ negative ‹ past"?
>
> &mdash; <cite>Me</cite>

**Comment by birtles** [comment]
Just a thought, would "polite negative past" work? Is there a need to actually separate the reasons at all (at least in the English localization).

**Reply by enellis** [reply]
I initially proposed this idea because, technically speaking, there is no form such as "polite past negative" or "polite negative past".

Just to explain my thought process, when we dissect 頑張りませんでした, for example:
‹ There is the 連用形 of 頑張る, which is 頑張り. To the 連用形 connects the auxiliary verb ます, making it polite.
‹ The irregular negative form of ます is ません, transforming the whole expression, simply speaking, into a noun.
‹ With nouns, to express the past, we use the perfective form of the copula でした.

So, breaking it down into `‹ polite ‹ negative ‹ past` captures the grammatical process more accurately than stating that it is the single form "polite negative past".

We also deinflect the plain past negative as `‹ negative ‹ past`, not `‹ negative past`.

However, you are right. We should aim for clarity and it should make sense for everyone immediately. Most learners are taught these forms as a set and don't break them down into smaller parts. It really might be best to keep those compounds as single forms.

I’d be happy to simply change it to "polite negative past". If we decide to implement this, I could incorporate this into the 敬語-PR.

**Reply by birtles** [reply]
Thanks! I think this change should be a separate PR. I prefer to keep the PRs fine-grained when possible and it would also allow more discussion and time to check that all the concatenations of reasons actually make sense.

**Reply by enellis** [reply]
Sure, there's no need to rush.
